### DISCUSSION TITLE
Get specific attribut of an AD user

### DISCUSSION THREAD

**Discussion by klepptor** [DirectoryTree/LdapRecord#481] [Q&A] [answered]
First of all: sorry for asking! The answer must be really easy to find, but I can't!

I'm trying to

1. search if a userPrincipalName exists in AD
2. if it exists look up the users distinguishedName
3. check if this dn is in the members attribut of some groups

What works
```
$user = new User();
$user->findBy('userPrincipalName','user@dom.int')
// returns true if found
// or:
$user = $connection->query()->where([
    ['userPrincipalName', '=', 'user@dom.int'],
])->first();
// returns the user as object

// don't know what's the better way!?
```

But what now?

There is the getAttribute() method (https://ldaprecord.com/docs/core/v2/model-api/#getattribute), but it doesn't exist on the returned user object but on these "Models" and I don't understand how they work....

Thx!

**Comment by stevebauman** [comment, marked as answer]
Hi @klepptor,

You appear to be using both raw queries on your connection and models. I'd suggest sticking with models instead of running raw queries.

This should work for you:

```php
use LdapRecord\Models\ActiveDirectory\User;
use LdapRecord\Models\ActiveDirectory\Group;

$user = User::findByOrFail('userPrincipalName, 'user@dom.int');

// $user->getDn(); // Get the users DN (if you need it).

$group = Group::findByOrFail('cn', 'Some Group');

if ($user->groups()->exists($group)) {
    // The user is a member of the group.
}
```

Here are the docs for user and group management for Active Directory:

https://ldaprecord.com/docs/core/v2/active-directory/users
https://ldaprecord.com/docs/core/v2/active-directory/groups

**Comment by klepptor** [comment]
Wow, so fast and so good! ;)

Thank you very much!!!

**Reply by stevebauman** [reply]
Happy to help @klepptor! 😄
